### ISSUE TITLE
API and implimentation separation ignored by `distZip` and `distTar` operations

### ISSUE THREAD

**Issue by leamingrad** [gradle/gradle#7380]
When using the `java-library-distribution` plugin, the `distZip` and `distTar` tasks ignore `api` dependencies instead of packaging them in the distribution.

### Expected Behavior
I would expect the `distZip` and `distTar` operations to include all dependencies required at runtime in the distribution. When using the `java-library` plugin, the [documentation](https://docs.gradle.org/current/userguide/java_library_plugin.html#sec:java_library_separation) says that API and implementation dependencies should be separated, but I would expect API dependencies to be required at runtime (and so bundled).

### Current Behavior
Currently, only dependencies that are used by the `runtime` configuration are bundled, which includes neither `implementation` or `api` dependencies.

### Context
This was hit when migrating our gradle files from using the `java` plugin to the `java-library` plugin (as we are actually shipping a library, and moving our dependencies to use the best-practice layout at the same time.

### Steps to Reproduce (for bugs)
The following minimum `build.gradle` will hit this issue:

```
plugins {
    id 'ivy-publish'
    id 'java-library'
    id 'java-library-distribution'
}

group = 'test'
version = '0.0.1'


dependencies {
    api 'com.google.code.gson:gson:2.3.1'

    testImplementation 'junit:junit:4.12'
    testImplementation 'org.hamcrest:hamcrest-library:1.3'
}

publishing {
    repositories {
        ivy {
	   url = "/tmp/local-ivy-repo/"
        }
    }
    publications {
        ivy(IvyPublication) {
            from components.java

	    artifact distTar
	    artifact distZip
        }
    }
}
```

### Your Environment
Output of `./gradlew --version`:

```
------------------------------------------------------------
Gradle 4.10.1
------------------------------------------------------------

Build time:   2018-09-12 11:33:27 UTC
Revision:     76c9179ea9bddc32810f9125ad97c3315c544919

Kotlin DSL:   1.0-rc-6
Kotlin:       1.2.61
Groovy:       2.4.15
Ant:          Apache Ant(TM) version 1.9.11 compiled on March 23 2018
JVM:          1.8.0_171 (Oracle Corporation 25.171-b11)
OS:           Linux 4.18.0-1-amd64 amd64
```

**Comment by thc202** [comment]
Seems to be a duplicate of #5695.

**Comment by oehme** [comment]
Yep, this is fixed for 5.0
